lighthouse-core/audits/dobetterweb/obscured-largest-contentful-paint.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit.js');
+const i18n = require('../../lib/i18n/i18n.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that */
+  title: 'Largest Contentful Paint element is fully visible',
+  /** Title of a Lighthouse audit that */
+  failureTitle: 'Avoid obscuring the Largest Contentful Paint element',
+  /** Description of a Lighthouse audit that  */
+  description: 'Description',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class ObscuredLargestContentfulPaint extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'obscured-largest-contentful-paint',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['traces', 'ObscuringElements'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    const obscuringElements = artifacts.ObscuringElements;
```

Comment:
presumably we want to include the node details in this audit's `details`?

Reply:
I was unsure about this which is why I left it pretty barebones. Do we want to show all of the elements that obscure the LCP or maybe just the top most container/element that covers the LCP? The latter will require some logic I opted to leave out until I was sure (finding common ancestors and what not).

Reply:
> Do we want to show all of the elements that obscure the LCP or maybe just the top most container/element that covers the LCP?

I still think this could be interesting. And perhaps necessary (there could possibly be so many...)?

Let's get some example pages before we decided on this. Maybe instead of relying on existing sites and figuring out how to trigger their interstitials, let's just make some of our own?

aside: it'd be super nice if we could kick off a GAE script to run Lighthouse at megascale for just a single audit-in-progress and allow us to view the distribution in results. Without a step sort of like that we are really just guessing ...